diff --git a/net/socket/connect_job_params.h b/net/socket/connect_job_params.h
index 12eb56af2f..b7bd811b03 100644
--- a/net/socket/connect_job_params.h
+++ b/net/socket/connect_job_params.h
@@ -13,6 +13,7 @@
 namespace net {
 
 class HttpProxySocketParams;
+class RspSocketParams;
 class SOCKSSocketParams;
 class TransportSocketParams;
 class SSLSocketParams;
@@ -22,6 +23,7 @@ class NET_EXPORT_PRIVATE ConnectJobParams {
  public:
   ConnectJobParams();
   explicit ConnectJobParams(scoped_refptr<HttpProxySocketParams> params);
+  explicit ConnectJobParams(scoped_refptr<RspSocketParams> params);
   explicit ConnectJobParams(scoped_refptr<SOCKSSocketParams> params);
   explicit ConnectJobParams(scoped_refptr<TransportSocketParams> params);
   explicit ConnectJobParams(scoped_refptr<SSLSocketParams> params);
@@ -37,6 +39,10 @@ class NET_EXPORT_PRIVATE ConnectJobParams {
         params_);
   }
 
+  bool is_rsp() const {
+    return std::holds_alternative<scoped_refptr<RspSocketParams>>(params_);
+  }
+
   bool is_socks() const {
     return std::holds_alternative<scoped_refptr<SOCKSSocketParams>>(params_);
   }
@@ -54,6 +60,9 @@ class NET_EXPORT_PRIVATE ConnectJobParams {
   const scoped_refptr<HttpProxySocketParams>& http_proxy() const {
     return get<scoped_refptr<HttpProxySocketParams>>(params_);
   }
+  const scoped_refptr<RspSocketParams>& rsp() const {
+    return get<scoped_refptr<RspSocketParams>>(params_);
+  }
   const scoped_refptr<SOCKSSocketParams>& socks() const {
     return get<scoped_refptr<SOCKSSocketParams>>(params_);
   }
@@ -68,6 +77,9 @@ class NET_EXPORT_PRIVATE ConnectJobParams {
   scoped_refptr<HttpProxySocketParams>&& take_http_proxy() {
     return get<scoped_refptr<HttpProxySocketParams>>(std::move(params_));
   }
+  scoped_refptr<RspSocketParams>&& take_rsp() {
+    return get<scoped_refptr<RspSocketParams>>(std::move(params_));
+  }
   scoped_refptr<SOCKSSocketParams>&& take_socks() {
     return get<scoped_refptr<SOCKSSocketParams>>(std::move(params_));
   }
@@ -80,6 +92,7 @@ class NET_EXPORT_PRIVATE ConnectJobParams {
 
  private:
   std::variant<scoped_refptr<HttpProxySocketParams>,
+               scoped_refptr<RspSocketParams>,
                scoped_refptr<SOCKSSocketParams>,
                scoped_refptr<TransportSocketParams>,
                scoped_refptr<SSLSocketParams>>
