diff --git a/chrome/browser/ui/views/toolbar/BUILD.gn b/chrome/browser/ui/views/toolbar/BUILD.gn
--- a/chrome/browser/ui/views/toolbar/BUILD.gn
+++ b/chrome/browser/ui/views/toolbar/BUILD.gn
@@ -28,6 +28,7 @@ source_set("toolbar") {
     "pinned_toolbar_actions_container_layout.h",
     "pinned_toolbar_button_status_indicator.h",
     "reload_button.h",
+    "rsp_toolbar_button.h",
     "reload_control.h",
     "split_tabs_button.h",
     "toolbar_action_hover_card_bubble_view.h",
@@ -116,6 +117,7 @@ source_set("impl") {
     "pinned_toolbar_actions_container_layout.cc",
     "pinned_toolbar_button_status_indicator.cc",
     "reload_button.cc",
+    "rsp_toolbar_button.cc",
     "split_tabs_button.cc",
     "toolbar_action_hover_card_bubble_view.cc",
     "toolbar_action_hover_card_controller.cc",
@@ -196,6 +198,7 @@ source_set("impl") {
     "//chrome/browser/profiles:profile",
     "//chrome/browser/profiles:profile_manager",
     "//chrome/browser/profiles:profile_util",
+    "//chrome/browser/rsp",
     "//chrome/browser/search",
     "//chrome/browser/sharing_hub",
     "//chrome/browser/signin",
