diff --git a/chrome/common/url_constants.h b/chrome/common/url_constants.h
index 0f9862d51b..41b31b321d 100644
--- a/chrome/common/url_constants.h
+++ b/chrome/common/url_constants.h
@@ -168,6 +168,9 @@ inline constexpr char kChromeSearchRemoteNtpHost[] = "remote-ntp";
 //     coming from a blessed Instant process, and deny the request.
 inline constexpr char kChromeSearchScheme[] = "chrome-search";
 
+// RSP protocol scheme.
+inline constexpr char kRspScheme[] = "rsp";
+
 // This is the base URL of content that can be embedded in chrome://new-tab-page
 // using an <iframe>. The embedded untrusted content can make web requests and
 // can include content that is from an external source.
