diff --git a/chrome/browser/ui/views/tabs/new_tab_button.cc b/chrome/browser/ui/views/tabs/new_tab_button.cc
index 67a7c1be1f..8b30f31e02 100644
--- a/chrome/browser/ui/views/tabs/new_tab_button.cc
+++ b/chrome/browser/ui/views/tabs/new_tab_button.cc
@@ -28,6 +28,7 @@
 DEFINE_CLASS_ELEMENT_IDENTIFIER_VALUE(NewTabButtonMenuModel, kNewTab);
 DEFINE_CLASS_ELEMENT_IDENTIFIER_VALUE(NewTabButtonMenuModel, kNewTabInGroup);
 DEFINE_CLASS_ELEMENT_IDENTIFIER_VALUE(NewTabButtonMenuModel, kNewSplitView);
+DEFINE_CLASS_ELEMENT_IDENTIFIER_VALUE(NewTabButtonMenuModel, kNewRspTab);
 DEFINE_CLASS_ELEMENT_IDENTIFIER_VALUE(NewTabButtonMenuModel,
                                       kCreateNewTabGroup);
 
@@ -84,6 +85,12 @@ NewTabButtonMenuModel::NewTabButtonMenuModel(BrowserWindowInterface* browser)
 
   AddSeparator(ui::NORMAL_SEPARATOR);
   AddNewSplitTabItem();
+
+
+  AddSeparator(ui::NORMAL_SEPARATOR);
+  AddItem(IDC_NEW_RSP_TAB, u"New RSP Tab");
+  SetElementIdentifierAt(
+      GetIndexOfCommandId(IDC_NEW_RSP_TAB).value(), kNewRspTab);
 }
 
 NewTabButtonMenuModel::~NewTabButtonMenuModel() = default;
@@ -108,6 +115,10 @@ void NewTabButtonMenuModel::ExecuteCommand(int command_id, int event_flags) {
       base::RecordAction(
           base::UserMetricsAction("NewTabButton_ContextMenu_NewSplitTab"));
       break;
+    case IDC_NEW_RSP_TAB:
+      base::RecordAction(
+          base::UserMetricsAction("NewTabButton_ContextMenu_NewRspTab"));
+      break;
   }
 
   if (command_id == IDC_NEW_SPLIT_TAB) {
@@ -118,6 +129,11 @@ void NewTabButtonMenuModel::ExecuteCommand(int command_id, int event_flags) {
     return;
   }
 
+  if (command_id == IDC_NEW_RSP_TAB) {
+    chrome::NewRspTab(browser_);
+    return;
+  }
+
   chrome::ExecuteCommand(browser_, command_id);
 }
 
