diff --git a/chrome/browser/ui/views/tabs/new_tab_button.h b/chrome/browser/ui/views/tabs/new_tab_button.h
index 5ea237a..e41e2df 100644
--- a/chrome/browser/ui/views/tabs/new_tab_button.h
+++ b/chrome/browser/ui/views/tabs/new_tab_button.h
@@ -55,6 +55,7 @@ class NewTabButtonMenuModel : public ui::SimpleMenuModel,
   DECLARE_CLASS_ELEMENT_IDENTIFIER_VALUE(kNewTab);
   DECLARE_CLASS_ELEMENT_IDENTIFIER_VALUE(kNewTabInGroup);
   DECLARE_CLASS_ELEMENT_IDENTIFIER_VALUE(kNewSplitView);
+  DECLARE_CLASS_ELEMENT_IDENTIFIER_VALUE(kNewRspTab);
   DECLARE_CLASS_ELEMENT_IDENTIFIER_VALUE(kCreateNewTabGroup);
   explicit NewTabButtonMenuModel(BrowserWindowInterface* browser);
   NewTabButtonMenuModel(const NewTabButtonMenuModel&) = delete;
