diff --git a/chrome/browser/ui/views/tabs/tab.h b/chrome/browser/ui/views/tabs/tab.h
index e3fe43ec48..d97773e05a 100644
--- a/chrome/browser/ui/views/tabs/tab.h
+++ b/chrome/browser/ui/views/tabs/tab.h
@@ -44,6 +44,7 @@ class Animation;
 class LinearAnimation;
 }  // namespace gfx
 namespace views {
+class ImageButton;
 class View;
 }  // namespace views
 
@@ -266,6 +267,8 @@ class Tab : public gfx::AnimationDelegate,
 
   void CloseButtonPressed(const ui::Event& event);
 
+  void RspProxyButtonPressed(const ui::Event& event);
+
   void OnTabDataChanged(TabChangeType tab_change_type,
                         const tabs::TabData& tab_data);
 
@@ -286,6 +289,7 @@ class Tab : public gfx::AnimationDelegate,
 
   raw_ptr<TabIcon> icon_ = nullptr;
   raw_ptr<AlertIndicatorButton> alert_indicator_button_ = nullptr;
+  raw_ptr<views::ImageButton> rsp_proxy_button_ = nullptr;
   raw_ptr<TabCloseButton> close_button_ = nullptr;
 
   raw_ptr<TabTitle> title_;
